static/src/javascripts-legacy/projects/common/modules/experiments/tests/tailor-survey.js

```diff
@@ -48,31 +50,124 @@ define([
             return !(config.page.isAdvertisementFeature) &&
                 config.page.contentType === 'Article'
         };
+        function callTailor(bwid, surveysNotShowAgain) {
+            // If we want to force tailor to show a particular survey we can set an attribute in local storage to have
+            // key = 'surveyToShow', and value = the survey id. Tailor will then override other logic for display, and
+            // look for a survey with this ID to return. This is useful as we can easily see how a particular survey
+            // would be rendered, without actually putting it live. If this parameter is empty or not specified, tailor
+            // behaves as usual.
+            var surveyToShow = localStorage.getItem('surveyToShow');
 
-        function callTailor(bwid) {
-            var endpoint = 'https://tailor.guardianapis.com/suggestions?browserId=' + bwid;
+            var endpoint = 'https://tailor.guardianapis.com/suggestions'+
+                '?browserId=' + bwid +
+                '&edition=' + config.page.edition +
+                '&surveyToShow=' + surveyToShow +
+                '&surveysNotToShow=' + surveysNotShowAgain;
             return fetchJson(endpoint, {
                 type: 'json',
                 method: 'get'
             });
         }
 
+        // Every time we show a survey to a user, we cannot show it again to that suer for a specified number of days.
+        // We store 'surveyId=dayShowAgain' in the cookie, and pass any surveys that cannot currently be shown in the
+        // call to tailor.
+        function storeSurveyShowedInCookie(surveySuggestionToShow) {
+            var id = surveySuggestionToShow.data.survey.surveyId;
+            var dayCanShowAgain = surveySuggestionToShow.data.dayCanShowAgain;
+
+            var newCookieValue = id + '=' + dayCanShowAgain;
+
+
+
+            var currentCookieValues = cookies.get('GU_TAILOR_SURVEY');
+
+            if (currentCookieValues) {
+                // we've shown surveys already
+                currentCookieValues = currentCookieValues + ',' + newCookieValue;
+                cookies.remove('GU_TAILOR_SURVEY');
+                cookies.add('GU_TAILOR_SURVEY', currentCookieValues, 365);
+            }
+            else {
+                // first time we show any survey
+                cookies.add('GU_TAILOR_SURVEY', newCookieValue, 365);
+            }
+        }
+        // Given a response from tailor, we see if the response has a survey suggestion, and if so return the first
+        // survey suggestion (there should only ever be one, but just in case).
+        function getSurveySuggestionToShow(response) {
+            if (response.suggestions) {
+                var surveySuggestions = response.suggestions.filter(function (suggestion) {
+                    return suggestion.class == 'SurveySuggestion';
+                });
+
+                if (surveySuggestions) {
```

Comment:
`filter` always returns an array, so `surveySuggestions` will always be truthy, even if the array is empty. It would be better to check `surveySuggestions.length`